cloud_utils/vault.py

```diff
@@ -1,136 +1,51 @@
-from typing import Any, Callable, Coroutine, Dict, Optional
+from typing import Callable, Optional
 
-import gamla
-from cache import AsyncTTL
+import hvac
+from hvac import exceptions
+from hvac.api.auth_methods import Kubernetes
 
-base_metadata_url = "http://169.254.169.254/metadata"
+_MOUNT_POINT = "secret"
 
 
-class KeyDoesNotExist(Exception):
-    pass
-
-
-async def _identity_token():
-    return (
-        (
-            await gamla.get_async_with_headers(
-                {"Metadata": "true"},
-                30,
-                f"{base_metadata_url}/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fmanagement.azure.com%2F",
-            )
+def _build_read_all_secrets(client: hvac.Client):
+    def read_all_secrets(path: str, version: Optional[str]) -> dict[str, str]:
+        read_secrets = client.secrets.kv.v2.read_secret_version(
+            path=path, version=version
         )
-        .json()
-        .get("access_token")
-    )
+        return read_secrets["data"]["data"]
 
+    return read_all_secrets
 
-async def _instance_metadata():
-    return (
-        await gamla.get_async_with_headers(
-            {"Metadata": "true"},
-            30,
-            f"{base_metadata_url}/instance?api-version=2019-08-15",
-        )
-    ).json()
 
-
-def _make_pod_identity_token(
-    role: str,
-    base_vault_url: str,
-) -> Callable[[], Coroutine[Any, Any, str]]:
-    @AsyncTTL(time_to_live=60 * 60 * 23, maxsize=1)  # Vault token is valid for 24 hour.
-    async def _pod_identity_token() -> str:
-        jwt = await _identity_token()
-        return await gamla.pipe(
-            await _instance_metadata(),
-            gamla.get_in(["compute"]),
-            lambda compute: {
-                "role": role,
-                "jwt": jwt,
-                "subscription_id": gamla.get_in(["subscriptionId"])(compute),
-                "resource_group_name": gamla.get_in(["resourceGroupName"])(compute),
-                "vm_name": gamla.get_in(["name"])(compute),
-                "vmss_name": gamla.get_in(["vmScaleSetName"])(compute),
-            },
-            gamla.post_json_async(30, f"{base_vault_url}/auth/azure/login"),
-            lambda response: response.json(),
-            gamla.get_in(["auth", "client_token"]),
+def _build_write_or_update_secrets(client: hvac.Client, read_all_secrets: Callable):
```

Comment:
`write_ot_update_secret` and the payload of the inner function is not "secrets" but a dictionary of keys and values (e.g. `new_keys`)